public/app/features/explore/state/query.ts

```diff
@@ -309,100 +327,125 @@ export const runQueries = (exploreId: ExploreId, options?: { replaceUrl?: boolea
       history,
       refreshInterval,
       absoluteRange,
+      cache,
     } = exploreItemState;
+    let newQuerySub;
+
+    const cachedValue = getResultsFromCache(cache, absoluteRange);
+
+    // If we have results saved in cache, we are going to use those results instead of running queries
+    if (cachedValue) {
+      newQuerySub = of(cachedValue)
+        .pipe(
+          map((data: PanelData) => preProcessPanelData(data, queryResponse)),
```

Comment:
Could we deduplicate pipe mappings? (same are used a few lines below)

Reply:
Yes, done as well in https://github.com/grafana/grafana/pull/34297/commits/a3af0a24d8c5159f2a25771cbcd9deecafa8dd5d.